Enforce salt Python linting

TeddyAndrieux opened the pull request:
**Component**:

'salt', 'lint', 'python'
<!-- E.g. 'salt', 'containers', 'kubernetes', 'build', 'tests'... -->

**Context**: 

#2649 

**Summary**:

- Fix python linting issue in all Salt modules
- Enable salt python linting in tox and doit, enforce it in CI build

---

Fixes: #2649

TeddyAndrieux committed:
tox: Add linting for Salt Python modules

Add a dedicated `pylintrc` file for all Salt Python modules and run a
pylint on all thoses files with tox `lint-python` as well as
`./doit.sh lint:python` so that we do some linting check in the CI

Fixes: #2649

TeddyAndrieux committed:
salt-lint: cri module fix "useless-else-on-loop"

```
salt/_modules/cri.py:174: [W0120(useless-else-on-loop), wait_container] Else clause on loop without a break statement
```

TeddyAndrieux committed:
salt-lint: metalk8s_cordon module fix "too-many-blank-lines"

```
salt/_modules/metalk8s_cordon.py:84: [E8303(too-many-blank-lines), ] PEP8 E303: too many blank lines (2)
```

TeddyAndrieux committed:
salt-lint: metalk8s_drain module fix "raise-missing-from"

Disable some "raise-missing-from" as it's known exception and we do not
want to have upper exceptions raised
```
salt/_modules/metalk8s_drain.py:388: [W0707(raise-missing-from), Drain.run_drain] Consider explicitly re-raising using the 'from' keyword
salt/_modules/metalk8s_drain.py:394: [W0707(raise-missing-from), Drain.run_drain] Consider explicitly re-raising using the 'from' keyword
```

Use `from` in exception instead of formatting with the exception
message
```
salt/_modules/metalk8s_drain.py:360: [W0707(raise-missing-from), Drain.run_drain] Consider explicitly re-raising using the 'from' keyword
salt/_modules/metalk8s_drain.py:547: [W0707(raise-missing-from), evict_pod] Consider explicitly re-raising using the 'from' keyword
```

TeddyAndrieux committed:
salt-lint: metalk8s_drain module fix "unecessary-pass"

Remove "unecessary-pass"
```
salt/_modules/metalk8s_drain.py:59: [W0107(unnecessary-pass), DrainTimeoutException] Unnecessary pass statement
```

TeddyAndrieux committed:
salt-lint: metalk8s_etcd module fix "blacklisted-module"

Use six urllib parse module from salt instead of six directly
```
salt/_modules/metalk8s_etcd.py:6: [E9402(blacklisted-module), ] Uses of a blacklisted module 'six.moves.urllib.parse': Please report this error to SaltStack so we can fix it: Trying to import urlparse from six.moves.urllib.parse
```

TeddyAndrieux committed:
salt-lint: metalk8s_etcd module fix "bare-except"

Fix "bare-except" and disable "broad-except" as we want to catch all
exceptions
```
salt/_modules/metalk8s_etcd.py:198: [W0702(bare-except), get_etcd_member_list] No exception type(s) specified
```

TeddyAndrieux committed:
salt-lint: metalk8s_kubeconfig fix "broad-except"

Disable "broad-except" as we want to catch all exceptions
```
salt/_modules/metalk8s_kubeconfig.py:43: [W0703(broad-except), validate] Catching too general exception Exception
```

TeddyAndrieux committed:
salt-lint: metalk8s_kubeconfig fix "resource-leakage"

Use `fopen` from salt instead of classic `open`
```
salt/_modules/metalk8s_kubeconfig.py:41: [W8470(resource-leakage), validate] Resource leakage detected. Please use 'with salt.utils.files.fopen()' instead of 'with open()'. It assures salt does not leak file handles.
```

TeddyAndrieux committed:
salt-lint: metalk8s_kubernetes module fix "raise-missing-from"

Use `from` in exception instead of formatting with the exception
message
```
salt/_modules/metalk8s_kubernetes.py:56: [W0707(raise-missing-from), _extract_obj_and_kind_info] Consider explicitly re-raising using the 'from' keyword
salt/_modules/metalk8s_kubernetes.py:110: [W0707(raise-missing-from), _object_manipulation_function.method] Consider explicitly re-raising using the 'from' keyword
salt/_modules/metalk8s_kubernetes.py:115: [W0707(raise-missing-from), _object_manipulation_function.method] Consider explicitly re-raising using the 'from' keyword
salt/_modules/metalk8s_kubernetes.py:345: [W0707(raise-missing-from), list_objects] Consider explicitly re-raising using the 'from' keyword
salt/_modules/metalk8s_kubernetes.py:374: [W0707(raise-missing-from), list_objects] Consider explicitly re-raising using the 'from' keyword
```

TeddyAndrieux committed:
salt-lint: metalk8s_kubernetes_utils module fix "raise-missing-from"

Use `from` in exception instead of formatting with the exception message
```
salt/_modules/metalk8s_kubernetes_utils.py:98: [W0707(raise-missing-from), get_version_info] Consider explicitly re-raising using the 'from' keyword
salt/_modules/metalk8s_kubernetes_utils.py:178: [W0707(raise-missing-from), get_service_endpoints] Consider explicitly re-raising using the 'from' keyword
salt/_modules/metalk8s_kubernetes_utils.py:201: [W0707(raise-missing-from), get_service_endpoints] Consider explicitly re-raising using the 'from' keyword
```

TeddyAndrieux committed:
salt-lint: metalk8s_monitoring module fix "raise-missing-from"

Use `from` in exception instead of formatting with the exception message
```
salt/_modules/metalk8s_monitoring.py:189: [W0707(raise-missing-from), _requests_alertmanager_api] Consider explicitly re-raising using the 'from' keyword
salt/_modules/metalk8s_monitoring.py:206: [W0707(raise-missing-from), _requests_alertmanager_api] Consider explicitly re-raising using the 'from' keyword
```

TeddyAndrieux committed:
salt-lint: metalk8s_package_manager_yum fix "unused-import"

Remove unused import
```
salt/_modules/metalk8s_package_manager_yum.py:7: [W0611(unused-import), ] Unused contextmanager imported from contextlib
```

TeddyAndrieux committed:
salt-lint: metalk8s module fix "blacklisted-module"

Use six module from salt instead of six directly
```
salt/_modules/metalk8s.py:10: [E9405(blacklisted-external-import), ] Uses of an external blacklisted import 'six': Please use 'import salt.ext.six as six'
```

TeddyAndrieux committed:
salt-lint: metalk8s module fix "raise-missing-from"

Use `from` in exception instead of formatting with the exception message
```
salt/_modules/metalk8s.py:102: [W0707(raise-missing-from), minions_by_role] Consider explicitly re-raising using the 'from' keyword
salt/_modules/metalk8s.py:348: [W0707(raise-missing-from), format_slots] Consider explicitly re-raising using the 'from' keyword
```

TeddyAndrieux committed:
salt-lint: metalk8s module fix "resource-leakage"

Use `fopen` from salt instead of classic `open`
```
salt/_modules/metalk8s.py:408: [W8470(resource-leakage), _atomic_copy] Resource leakage detected. Please use 'with salt.utils.files.fopen()' instead of 'with open()'. It assures salt does not leak file handles.
```

TeddyAndrieux committed:
salt-lint: metalk8s_service_configuration fix "raise-missing-from"

Use `from` in exception instead of formatting with the exception message
```
salt/_modules/metalk8s_service_configuration.py:67: [W0707(raise-missing-from), get_service_conf] Consider explicitly re-raising using the 'from' keyword
salt/_modules/metalk8s_service_configuration.py:81: [W0707(raise-missing-from), get_service_conf] Consider explicitly re-raising using the 'from' keyword
salt/_modules/metalk8s_service_configuration.py:87: [W0707(raise-missing-from), get_service_conf] Consider explicitly re-raising using the 'from' keyword
```

TeddyAndrieux committed:
salt-lint: metalk8s_solutions_k8s module fix "unused-import"

Remove unused import
```
salt/_modules/metalk8s_solutions_k8s.py:9: [W0611(unused-import), ] Unused CommandExecutionError imported from salt.exceptions
```

TeddyAndrieux committed:
salt-lint: metalk8s_solutions module fix "resource-leakage"

Use `fopen` from salt instead of classic `open`
```
salt/_modules/metalk8s_solutions.py:40: [W8470(resource-leakage), _load_config_file] Resource leakage detected. Please use 'with salt.utils.files.fopen()' instead of 'with open()'. It assures salt does not leak file handles.
salt/_modules/metalk8s_solutions.py:51: [W8470(resource-leakage), _write_config_file] Resource leakage detected. Please use 'with salt.utils.files.fopen()' instead of 'with open()'. It assures salt does not leak file handles.
```

TeddyAndrieux committed:
salt-lint: metalk8s_solutions module fix "raise-missing-from"

Use `from` in exception instead of formatting with the exception message
```
salt/_modules/metalk8s_solutions.py:46: [W0707(raise-missing-from), _load_config_file] Consider explicitly re-raising using the 'from' keyword
salt/_modules/metalk8s_solutions.py:57: [W0707(raise-missing-from), _write_config_file] Consider explicitly re-raising using the 'from' keyword
salt/_modules/metalk8s_solutions.py:328: [W0707(raise-missing-from), manifest_from_iso] Consider explicitly re-raising using the 'from' keyword
```

TeddyAndrieux committed:
salt-lint: metalk8s_sysctl module fix "unused-import"

Remove unused import
```
salt/_modules/metalk8s_sysctl.py:7: [W0611(unused-import), ] Unused import itertools
```

TeddyAndrieux committed:
salt-lint: metalk8s_sysctl module fix "raise-missing-from"

Disable "raise-missing-from" as it's known exception and we do not want
to have upper exceptions raised
```
salt/_modules/metalk8s_sysctl.py:78: [W0707(raise-missing-from), has_precedence] Consider explicitly re-raising using the 'from' keyword
salt/_modules/metalk8s_sysctl.py:84: [W0707(raise-missing-from), has_precedence] Consider explicitly re-raising using the 'from' keyword
```

TeddyAndrieux committed:
salt-lint: metalk8s_sysctl module fix "resource-leakage"

Use `fopen` from salt instead of classic `open`
```
salt/_modules/metalk8s_sysctl.py:94: [W8470(resource-leakage), has_precedence] Resource leakage detected. Please use 'with salt.utils.files.fopen()' instead of 'with open()'. It assures salt does not leak file handles.
```

TeddyAndrieux committed:
salt-lint: metalk8s_volumes module "2 spaces before inline comment"

Add 2 spaces before inline comments
```
salt/_modules/metalk8s_volumes.py:366: [E8261(at-least-two-spaces-before-inline-comment), ] PEP8 E261: at least two spaces before inline comment
salt/_modules/metalk8s_volumes.py:404: [E8261(at-least-two-spaces-before-inline-comment), ] PEP8 E261: at least two spaces before inline comment
```

TeddyAndrieux committed:
salt-lint: metalk8s_volumes module fix "2-blank-lines"

Add one more blank line between different functions
```
salt/_modules/metalk8s_volumes.py:371: [E8302(expected-2-blank-lines-found-0), ] PEP8 E302: expected 2 blank lines, found 1
salt/_modules/metalk8s_volumes.py:611: [E8302(expected-2-blank-lines-found-0), ] PEP8 E302: expected 2 blank lines, found 1
```

TeddyAndrieux committed:
salt-lint: metalk8s_volumes module fix "backslash-is-redundant-"

Re-format to not use `backslash` inside brackets
```
salt/_modules/metalk8s_volumes.py:250: [E8502(the-backslash-is-redundant-between-brackets), ] PEP8 E502: the backslash is redundant between brackets
```

TeddyAndrieux committed:
salt-lint: metalk8s_volumes module fix "blacklisted-module"

Use six module from salt instead of six directly
```
salt/_modules/metalk8s_volumes.py:14: [E9405(blacklisted-external-import), ] Uses of an external blacklisted import 'six': Please use 'import salt.ext.six as six'
```

TeddyAndrieux committed:
salt-lint: metalk8s_volumes module fix "unused-import"

Remove unused import
```
salt/_modules/metalk8s_volumes.py:7: [W0611(unused-import), ] Unused import fcntl
```

TeddyAndrieux committed:
salt-lint: metalk8s_volumes module fix "raise-missing-from"

Use `from` in exception instead of formatting with the exception message
```
salt/_modules/metalk8s_volumes.py:300: [W0707(raise-missing-from), SparseLoopDevice.create] Consider explicitly re-raising using the 'from' keyword
```

Disable "raise-missing-from" as it's known exception and we do not want
to have upper exceptions raised
```
salt/_modules/metalk8s_volumes.py:591: [W0707(raise-missing-from), _mkfs] Consider explicitly re-raising using the 'from' keyword
```

TeddyAndrieux committed:
salt-lint: containerd state fix "too-many-blank-lines"

Remove extra blank lines
```
salt/_states/containerd.py:43: [E8303(too-many-blank-lines), ] PEP8
E303: too many blank lines (2)
```

TeddyAndrieux committed:
salt-lint: metalk8s_kubernetes state fix "unused-import"

Remove unused import
```
salt/_states/metalk8s_kubernetes.py:10: [W0611(unused-import), ] Unused yaml imported from salt.utils
```

TeddyAndrieux committed:
salt-lint: metalk8s state fix "undefined-variable"

Add missing import for `CommandExecutionError` exception
```
salt/_states/metalk8s.py:81: [E0602(undefined-variable), static_pod_managed] Undefined variable 'CommandExecutionError'
```

TeddyAndrieux committed:
salt-lint: metalk8s_volumes state fix "broad-except"

Disable "broad-except" as we want to catch all exceptions
```
salt/_states/metalk8s_volumes.py:43: [W0703(broad-except), present] Catching too general exception Exception
salt/_states/metalk8s_volumes.py:78: [W0703(broad-except), prepared] Catching too general exception Exception
salt/_states/metalk8s_volumes.py:112: [W0703(broad-except), removed] Catching too general exception Exception
```

TeddyAndrieux committed:
salt-lint: metalk8s_volumes state fix "unused-import"

Remove unused import
```
salt/_states/metalk8s_volumes.py:6: [W0611(unused-import), ] Unused CommandExecutionError imported from salt.exceptions
```

TeddyAndrieux committed:
salt-lint: kubernetes_utils utils fix "arguments-differ"

Rename `method` to `verb` in initial class so that arguments name match
```
salt/_utils/kubernetes_utils.py:413: [W0221(arguments-differ), CustomApiClient._method] Parameters differ from overridden '_method' method
```

TeddyAndrieux committed:
salt-lint: kubernetes_utils utils fix "raise-missing-from"

Use `from` in exception instead of formatting with the exception message
```
salt/_utils/kubernetes_utils.py:694: [W0707(raise-missing-from), _build_standard_object] Consider explicitly re-raising using the 'from' keyword
```

Disable "raise-missing-from" as it's known exception and we do not want
to have upper exceptions raised
```
salt/_utils/kubernetes_utils.py:564: [W0707(raise-missing-from), _DictWrapper.__getattribute__] Consider explicitly re-raising using the 'from' keyword
salt/_utils/kubernetes_utils.py:622: [W0707(raise-missing-from), get_kind_info] Consider explicitly re-raising using the 'from' keyword
salt/_utils/kubernetes_utils.py:754: [W0707(raise-missing-from), _cast_value] Consider explicitly re-raising using the 'from' keyword
salt/_utils/kubernetes_utils.py:783: [W0707(raise-missing-from), _cast_value] Consider explicitly re-raising using the 'from' keyword
```

TeddyAndrieux committed:
salt-lint: pillar_utils utils fix "expected-2-blank-lines"

Add one more blank line
```
salt/_utils/pillar_utils.py:8: [E8302(expected-2-blank-lines-found-0), ] PEP8 E302: expected 2 blank lines, found 1
```

TeddyAndrieux committed:
salt-lint: volume_utils utils fix "backslash-is-redundant"

Re-format to not use `backslash` inside brackets
```
salt/_utils/volume_utils.py:280: [E8502(the-backslash-is-redundant-between-brackets), ] PEP8 E502: the backslash is redundant between brackets
```

TeddyAndrieux committed:
salt-lint: metalk8s_etcd pillar fix "broad-except"

Disable "broad-except" as we want to catch all exceptions
```
salt/_pillar/metalk8s_etcd.py:18: [W0703(broad-except), _load_members] Catching too general exception Exception
```

TeddyAndrieux committed:
salt-lint: metalk8s_private pillar fix "resource-leakage"

Use `fopen` from salt instead of classic `open`
```
salt/_pillar/metalk8s_private.py:19: [W8470(resource-leakage), _read_private_key] Resource leakage detected. Please use 'with salt.utils.files.fopen()' instead of 'with open()'. It assures salt does not leak file handles.
```

TeddyAndrieux committed:
salt-lint: metalk8s pillar fix "missing-whitspace-after-comma"

Remove uneeded comma
```
salt/_pillar/metalk8s.py:187: [E8231(missing-whitespace-after-comma), ] PEP8 E231: missing whitespace after ','
```

TeddyAndrieux committed:
salt-lint: metalk8s pillar fix "broad-except"

Disable "broad-except" as we want to catch all exceptions
```
salt/_pillar/metalk8s.py:28: [W0703(broad-except), _load_config] Catching too general exception Exception
```

TeddyAndrieux committed:
salt-lint: metalk8s_solutions pillar fix "broad-except"

Disable "broad-except" as we want to catch all exceptions
```
salt/_pillar/metalk8s_solutions.py:43: [W0703(broad-except), _load_solutions] Catching too general exception Exception
salt/_pillar/metalk8s_solutions.py:50: [W0703(broad-except), _load_solutions] Catching too general exception Exception
salt/_pillar/metalk8s_solutions.py:70: [W0703(broad-except), _load_solutions] Catching too general exception Exception
```

TeddyAndrieux committed:
salt-lint: metalk8s_saltutil runner fix "undefined-variable"

Add missing import for `SaltInvocationError` exception
```
salt/_runners/metalk8s_saltutil.py:136: [E0602(undefined-variable), orchestrate_show_sls] Undefined variable 'SaltInvocationError'
```

TeddyAndrieux committed:
salt-lint: kubernetes_rbac auth fix "expected-2-blank-lines"

Add one more blank line
```
salt/_auth/kubernetes_rbac.py:24: [E8302(expected-2-blank-lines-found-0), ] PEP8 E302: expected 2 blank lines, found 1
```

TeddyAndrieux committed:
salt-lint: kubernetes_rbac auth fix "unused-import"

Remove unused import
```
salt/_auth/kubernetes_rbac.py:1: [W0611(unused-import), ] Unused import base64
salt/_auth/kubernetes_rbac.py:10: [W0611(unused-import), ] Unused ApiException imported from kubernetes.client.rest
```

Disable "unused-import" as we import to check for virtual
```
salt/_auth/kubernetes_rbac.py:15: [W0611(unused-import), ] Unused import requests
```

TeddyAndrieux committed:
salt-lint: kubernetes_rbac auth fix "undefined-variable"

Add missing import for `CommandExecutionError` exception
```
salt/_auth/kubernetes_rbac.py:57: [E0602(undefined-variable), _check_auth_args.wrapped] Undefined variable 'CommandExecutionError'
```

TeddyAndrieux committed:
salt-lint: metalk8s_kubeconfig_info beacons fix "blacklisted-module"

Use six module from salt instead of six directly
```
salt/_beacons/metalk8s_kubeconfig_info.py:4: [E9405(blacklisted-external-import), ] Uses of an external blacklisted import 'six': Please use 'import salt.ext.six as six'
```

TeddyAndrieux committed:
salt-lint: metalk8s_kubernetes renderers fix "raise-missing-from"

Use `from` in exception instead of formatting with the exception message
```
salt/_renderers/metalk8s_kubernetes.py:41: [W0707(raise-missing-from), _step_name] Consider explicitly re-raising using the 'from' keyword
```

bert-e commented:
# Hello teddyandrieux,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @jbertran

* @chengyanjin

* @slaperche-scality

* @thomasdanan

* @lucieleonard

* @gdemonet

* @teddyandrieux

* @nootal

* @monpote

* @tcarmet

* @sayf-eddine-scality

* @cmonfort

* @nicolast

* @carlito-scality

* @alexandre-allard-scality

NicolasT commented on salt/_modules/metalk8s_drain.py:
```diff
@@ -56,7 +56,7 @@ def __str__(self):
 class DrainTimeoutException(DrainException):
     '''Timeout-specific drain exception.'''
-    pass
+    pass  # pylint: disable=unnecessary-pass
```
Can't it be removed instead?

TeddyAndrieux replied:
It could, but maybe we want to keep this one for code clarity, no strong opinion on this one, with a `pass` or without both ok for me

gdemonet replied:
From what I've seen in many Python codebases, the `pass` is usually omitted when possible (like in this case, because there's a docstring).
Exception subclasses are often using this approach: see https://github.com/saltstack/salt/blob/master/salt/exceptions.py#L72-L117 for instance.
So IMHO, we should remove it and let pylint check that we remain consistent.

TeddyAndrieux replied:
Ok fair I remove this one

NicolasT commented on salt/_modules/metalk8s_drain.py:
```diff
@@ -357,7 +357,7 @@ def run_drain(self, dry_run=False):
             pods = self.get_pods_for_eviction()
         except DrainException as exc:
-            raise CommandExecutionError(
+            raise CommandExecutionError(  # pylint: disable=raise-missing-from
```
Can't the `from` be added instead?

TeddyAndrieux replied:
We can but not sure it bring any value, you will just get the traceback to see from where it gets raised and it's will likely be on line 
```
        if failures:
            raise DrainException(_message_from_pods_dict(failures))
```
So ....

gdemonet replied:
What if we later change the `DrainException` to be raised from another one? IMO, if `pylint` enforces it as a rule, a `raise Something from caught_exception` never hurts.

TeddyAndrieux replied:
Ok fair

NicolasT commented on salt/_modules/metalk8s_monitoring.py:
```diff
@@ -203,7 +203,7 @@ def _requests_alertmanager_api(route, method='GET', **kwargs):
                 .format(exc, response.text)
             )
-        raise CommandExecutionError(error)
+        raise CommandExecutionError(error)  # pylint: disable=raise-missing-from
```
Same?

TeddyAndrieux replied:
Do we really want to raise the ValueError part of `response.json` ? 
Hmm maybe it make sense so that we know what the problem with the response content as it may not be obvious even if you see the response text
Ok fair I will update this one

gdemonet commented on salt/tests/unit/modules/files/test_metalk8s_kubernetes.yaml:
```diff
@@ -11,7 +11,7 @@ common_tests:
     info_scope: null
     raises: True
-    result: "Invalid manifest: An error has occurred"
+    result: "Invalid manifest"
```
In tests, could be useful to check that we can find the original error in the traceback. Maybe to tackle later, as debt.

TeddyAndrieux replied:
Hmm you right, I will try to do something about it

TeddyAndrieux replied:
Ok I think we will do it later I will open a ticket for this, basically you need to check the content of `__cause__` in the exception so will likely need to create a new `assertRaises...` to handle it (#3086)

gdemonet replied:
Perfect, thanks!

gdemonet commented on salt/_states/metalk8s.py:
```diff
@@ -5,6 +5,7 @@
 import traceback
 import re
 
+from salt.exceptions import CommandExecutionError
```
:+1: (undefined vars are a really useful check :heart:)

gdemonet commented on salt/_renderers/metalk8s_kubernetes.py:
```diff
@@ -38,7 +38,7 @@ def _step_name(manifest, absent=False):
         name = manifest['metadata']['name']
     except KeyError:
-        raise SaltRenderError('Object `metadata.name` must be set.')
+        raise SaltRenderError('Object `metadata.name` must be set.')  # pylint: disable=raise-missing-from
```
This one could have been kept (just to know whether the missing key is `metadata` or `name`)

TeddyAndrieux replied:
Right, make sense

gdemonet commented on salt/.pylintrc:
```diff
@@ -0,0 +1,704 @@
+  too-many-lines,
+  bad-continuation,
+  exec-used,
+  attribute-defined-outside-init,
+  protected-access,
+  reimported,
+  fixme,
+  global-statement,
+  unused-variable,
+  unused-argument,
+  redefined-outer-name,
+  redefined-builtin,
+  undefined-loop-variable,
+  logging-format-interpolation,
+  invalid-format-index,
+  line-too-long,
+  unexpected-indentation-comment,
+  continuation-line-indentation-is-not-a-multiple-of-four,
+  continuation-line-missing-indentation-or-outdented,
+  closing-bracket-does-not-match-indentation-of-opening-brackets-line,
+  closing-bracket-does-not-match-visual-indentation,
+  continuation-line-does-not-distinguish-itself-from-next-logical-line,
+  continuation-line-over-indented-for-hanging-indent,
+  continuation-line-over-indented-for-visual-indent,
+  continuation-line-under-indented-for-visual-indent,
+  visually-indented-line-with-same-indent-as-next-logical-line,
+  unaligned-for-hanging-indent,
+  block-comment-should-start-with-cardinal-space,
+  too-many-leading-hastag-for-block-comment,
+  module-level-import-not-at-top-of-file,
+  do-not-assign-a-lambda-expression-use-a-def,
+  3rd-party-local-module-not-gated,
+  pep8-reserved-keywords,
+  str-format-in-logging,
+  import-outside-toplevel,
+  deprecated-method,
+  repr-flag-used-in-string,
+  keyword-arg-before-vararg,
+  incompatible-py3-code,
+  multiple-spaces-before-operator
```
Isn't this a lot to disable? Recommended by Salt? Some checks like `unused-variable`/`undefined-loop-variable` may be useful, don't you think?

TeddyAndrieux replied:
Yes maybe, btw this one just get copied from pylintrc from salt repo, but indeed we may want to tune it a bit (I remember I already did some minor changes, do not remember the exact ones)

gdemonet replied:
OK let's keep it like this for now, and add some debt ticket to track that we'll want to include more rules :)

TeddyAndrieux replied:
I will remove from ignore list (and fix our code if needed) for at least those 
```
whitespace-before-colon,
indentation-is-not-a-multiple-of-four-comment,
whitespace-before-colon,
indentation-is-not-a-multiple-of-four-comment,
undefined-loop-variable,
unexpected-indentation-comment,
continuation-line-indentation-is-not-a-multiple-of-four,
continuation-line-missing-indentation-or-outdented,
closing-bracket-does-not-match-indentation-of-opening-brackets-line,
closing-bracket-does-not-match-visual-indentation,
continuation-line-does-not-distinguish-itself-from-next-logical-line,
continuation-line-over-indented-for-hanging-indent,
continuation-line-over-indented-for-visual-indent,
continuation-line-under-indented-for-visual-indent,
visually-indented-line-with-same-indent-as-next-logical-line,
unaligned-for-hanging-indent,
block-comment-should-start-with-cardinal-space,
too-many-leading-hastag-for-block-comment,
pep8-reserved-keywords,
str-format-in-logging,
```
Tell me if you have others that you think we should not ignore (anyway we could remove then from ignore list  in the future if needed)

gdemonet reviewed (changes requested):
Thanks for taking care of this, will be a good protection against simple mistakes! I just answered on some conversations, nothing really blocking though.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @thomasdanan

* @alexandre-allard-scality

* @jbwatenbergscality

* @gdemonet

* @lucieleonard

* @chengyanjin

* @teddyandrieux

* @nicolast





The following reviewers are expecting changes from the author, or must review again:

* @gdemonet

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @thomasdanan

* @alexandre-allard-scality

* @jbwatenbergscality

* @gdemonet

* @lucieleonard

* @chengyanjin

* @teddyandrieux

* @nicolast

TeddyAndrieux committed:
salt-lint: metalk8s module fix "unused-variable"

Remove the unused variable
```
salt/_modules/metalk8s.py:382: [W0612(unused-variable), _atomic_write] Unused variable 'base_name'
```

TeddyAndrieux committed:
salt-lint: metalk8s_drain module fix "unused-variable"

Remove the unused variable
```
salt/_modules/metalk8s_drain.py:516: [W0612(unused-variable), evict_pod] Unused variable 'result'
```

TeddyAndrieux committed:
salt-lint: metalk8s_package_manager_yum module fix "str-format-in-log"

Replace the `format` inside logging by a logging formatting
```
salt/_modules/metalk8s_package_manager_yum.py:51: [E1323(str-format-in-logging), _list_dependents] String formatting used in logging: "'{}-{}'.format(name, version)"
```

TeddyAndrieux committed:
salt-lint: metalk8s module fix "str-format-in-logging"

Replace the `format` inside logging by a logging formatting
```
salt/_modules/metalk8s.py:473: [E1323(str-format-in-logging), manage_static_pod_manifest] String formatting used in logging: "'Cached source file {} does not match expected checksum, will fetch it again'.format(source_filename)"
```

TeddyAndrieux committed:
salt-lint: metalk8s_service_configuration module fix "unused-argument"

Remove the unused argument
```
salt/_modules/metalk8s_service_configuration.py:20: [W0613(unused-argument), get_service_conf] Unused argument 'kwargs'
```

TeddyAndrieux committed:
salt-lint: metalk8s_volumes module fix "str-format-in-logging"

Replace the `format` inside logging by a logging formatting
```
salt/_modules/metalk8s_volumes.py:318: [E1323(str-format-in-logging), SparseLoopDevice.clean_up] String formatting used in logging: "'{} already removed'.format(exn.filename)"
```

TeddyAndrieux committed:
salt-lint: kubeconfig state fix "unused-argument"

Remove the unused argument
```
salt/_states/kubeconfig.py:12: [W0613(unused-argument), managed] Unused argument 'kwargs'
```

TeddyAndrieux committed:
salt-lint: metalk8s_endpoints pillar fix "unused-argument"

Disable "unused-argument" for `ext_pillar` as we need a specific
function signature
```
salt/_pillar/metalk8s_endpoints.py:20: [W0613(unused-argument), ext_pillar] Unused argument 'minion_id'
salt/_pillar/metalk8s_endpoints.py:20: [W0613(unused-argument), ext_pillar] Unused argument 'pillar'
```

TeddyAndrieux committed:
salt-lint: metalk8s_etcd pillar fix "unused-argument"

Disable "unused-argument" from `ext_pillar` as we need a specific
function signature
```
salt/_pillar/metalk8s_etcd.py:32: [W0613(unused-argument), ext_pillar] Unused argument 'minion_id'
```

TeddyAndrieux committed:
salt-lint: metalk8s pillar fix "unused-argument"

Disable "unused-argument" for `ext_pillar` as we need a specific
function signature
```
salt/_pillar/metalk8s.py:156: [W0613(unused-argument), ext_pillar] Unused argument 'minion_id'
salt/_pillar/metalk8s.py:156: [W0613(unused-argument), ext_pillar] Unused argument 'pillar'
```

TeddyAndrieux committed:
salt-lint: kubernetes_rbac auth fix "unused-argument"

Disable "unused-argument" for `auth` and `groups` as we need a specific
function signature
```
salt/_auth/kubernetes_rbac.py:198: [W0613(unused-argument), groups] Unused argument 'password'
```

Discard `kwargs` argument as it does not change the function signature
```
salt/_auth/kubernetes_rbac.py:180: [W0613(unused-argument), auth] Unused argument 'kwargs'
salt/_auth/kubernetes_rbac.py:198: [W0613(unused-argument), groups] Unused argument 'kwargs'
```

TeddyAndrieux committed:
salt-lint: metalk8s_solutions pillar fix "unused-argument"

Disable "unused-argument" for `ext_pillar` as we need a specific
function signature
```
salt/_pillar/metalk8s_solutions.py:81: [W0613(unused-argument), ext_pillar] Unused argument 'minion_id'
```

TeddyAndrieux committed:
salt-lint: metalk8s_kubernetes renderers fix "unused-argument"

Disable "unused-argument" for `render` as we need a specific function
signature
```
salt/_renderers/metalk8s_kubernetes.py:73: [W0613(unused-argument), render] Unused argument 'saltenv'
salt/_renderers/metalk8s_kubernetes.py:73: [W0613(unused-argument), render] Unused argument 'sls'
```

Discard `kwargs` argument as it does not change the function signature
```
salt/_renderers/metalk8s_kubernetes.py:73: [W0613(unused-argument), render] Unused argument 'kwargs'
```

TeddyAndrieux committed:
salt-lint: kubernetes_nodes roster fix "unused-argument"

Discard `kwargs` argument as it does not change the function signature
```
salt/_roster/kubernetes_nodes.py:14: [W0613(unused-argument), targets] Unused argument 'kwargs'
```

gdemonet commented on salt/_auth/kubernetes_rbac.py:
```diff
@@ -177,7 +177,7 @@ def _load_kubeconfig(opts):
 
 @_check_auth_args
-def auth(username, token=None, **kwargs):
+def auth(username, token=None, **kwargs):  # pylint: disable=unused-argument
```
I'm not sure, but I think there is some convention to describe "sink" arguments (ones that you will not use) to prefix them with an underscore.
How does pylint behave if you write instead:
```python
def auth(username, token=None, **_kwargs):
```
?

TeddyAndrieux replied:
For kwargs yes it's true we can do this. I will change this. 

Note that pylint point of view if you do a `def toto(_b=None)` for a `b` args that you do not use, pylint will report no error but then the signature of the function will be different

gdemonet reviewed (dismissed):
Just a style question on some function signatures, where we decide not to use some arguments.
Would you prefer keeping the original signature, as is done today, or make it explicit that you discard some arguments with `(..., **_kwargs):`?
Anyhow, let's not block the merge for this :wink:

TeddyAndrieux commented:
/approve

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @thomasdanan

* @alexandre-allard-scality

* @jbwatenbergscality

* @gdemonet

* @lucieleonard

* @chengyanjin

* @teddyandrieux

* @nicolast










*The following options are set:* **approve**

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.8`



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`
* `development/2.7`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.8`



The following branches have **NOT** changed:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`
* `development/2.7`



Please check the status of the associated issue None.

Goodbye teddyandrieux.
